src/clue/adafruit_slideshow.py

```diff
@@ -0,0 +1,303 @@
+                    )
+                    black_overlay.paste(new_img)
+                    new_img = black_overlay
+
+                black_overlay = Image.new("RGBA", new_img.size)
+                advance_sucessful = True
+            except Image.UnidentifiedImageError as e:
+                pass
+
+        # fade out old photo
+        for i in range(self.fade_frames, -1, -1):
+            sendable_img = Image.blend(
+                black_overlay, old_img, i * self.brightness / self.fade_frames
+            )
+            self._send(sendable_img)
+
+        time.sleep(self._BASE_DWELL_DARK)
+
+        # fade in new photo
+        for i in range(self.fade_frames + 1):
+            sendable_img = Image.blend(
+                black_overlay, new_img, i * self.brightness / self.fade_frames
+            )
+            self._send(sendable_img)
+
+        self._curr_img_handle = new_img
+        self._curr_img = new_path
+        self._img_start = time.monotonic()
+        return True
+
+    def _advance_no_fade(self):
+
+        old_img = self._curr_img_handle
+
+        advance_sucessful = False
+
+        while not advance_sucessful:
+            new_path = self._get_next_img()
+            if new_path == "":
+                return False
+
+            try:
+                new_img = Image.open(new_path)
+
+                new_img = new_img.crop(
+                    (0, 0, CONSTANTS.SCREEN_HEIGHT_WIDTH, CONSTANTS.SCREEN_HEIGHT_WIDTH)
+                )
+
+                if new_img.size[0] < 240 or new_img.size[1] < 240:
+                    black_overlay = Image.new(
+                        "RGBA",
+                        CONSTANTS.SCREEN_HEIGHT_WIDTH,
+                        CONSTANTS.SCREEN_HEIGHT_WIDTH,
+                    )
+                    black_overlay.paste(new_img)
+                    new_img = black_overlay
+
+                self._curr_img = new_path
+
+                new_img = new_img.convert("RGBA")
+                new_img.putalpha(255)
+                advance_sucessful = True
+            except Image.UnidentifiedImageError as e:
+                pass
+
+        if self.brightness < 1.0:
+            black_overlay = Image.new("RGBA", new_img.size)
+            new_img = Image.blend(black_overlay, new_img, self.brightness)
+
+        # gradually scroll new img over old img
+        for i in range(self._NO_FADE_TRANSITION_INCREMENTS + 1):
+            curr_y = (
+                i * CONSTANTS.SCREEN_HEIGHT_WIDTH / self._NO_FADE_TRANSITION_INCREMENTS
+            )
+            img_piece = new_img.crop((0, 0, CONSTANTS.SCREEN_HEIGHT_WIDTH, curr_y))
+            old_img.paste(img_piece)
+            self._send(old_img)
+
+        self._curr_img_handle = new_img
+        self._curr_img = new_path
+        self._img_start = time.monotonic()
+        return True
+
+    def _send(self, img):
+        # sends current bmp_img to the frontend
+        buffered = BytesIO()
+        img.save(buffered, format="BMP")

```

Comment:
Should we create const values for these "BMP" and ".bmp"? In case we change that in the future.